eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/OfferTypes.scala

```diff
@@ -38,27 +39,112 @@ object OfferTypes {
+   */
+  case class OfferMetadata(data: ByteVector) extends OfferTlv
+
+  /**
+   * Three-letter code of the currency the offer is denominated in. If empty, bitcoin is implied.
+   */
+  case class OfferCurrency(iso4217: String) extends OfferTlv
+
+  /**
+   * Amount to pay per item. As we only support bitcoin, the amount is in msat.
+   */
+  case class OfferAmount(amount: MilliSatoshi) extends OfferTlv
+
+  /**
+   * Description of the purpose of the payment.
+   */
+  case class OfferDescription(description: String) extends OfferTlv
+
+  /**
+   * Features supported to pay the offer.
+   */
+  case class OfferFeatures(features: Features[Feature]) extends OfferTlv
+
+  /**
+   * Time after which the offer is no longer valid.
+   */
+  case class OfferAbsoluteExpiry(absoluteExpiry: TimestampSecond) extends OfferTlv
+
+  /**
+   * Paths that can be used to retrieve an invoice.
+   */
+  case class OfferPaths(paths: Seq[BlindedRoute]) extends OfferTlv
+
+  /**
+   * Name of the offer creator.
+   */
+  case class OfferIssuer(issuer: String) extends OfferTlv
+
+  /**
+   * If present, the item described in the offer can be purchased multiple times with a single payment.
+   * If max = 0, there is no limit on the quantity that can be purchased in a single payment.
+   * If max > 1, it corresponds to the maximum number of items that be purchased in a single payment.
+   */
+  case class OfferQuantityMax(max: Long) extends OfferTlv
+
+  /**
+   * Public key of the offer creator.
+   * If `OfferPaths` is present, they must be used to retrieve an invoice even if this public key corresponds to a node id in the public network.
+   * If `OfferPaths` is not present, this public key must correspond to a node id in the public network that needs to be contacted to retrieve an invoice.
+   */
+  case class OfferNodeId(publicKey: PublicKey) extends OfferTlv
+
+  /**
+   * Random data to provide enough entropy so that some fields of the invoice request / invoice can be revealed without revealing the others.
+   */
+  case class InvoiceRequestMetadata(data: ByteVector) extends InvoiceRequestTlv
+
+  /**
+   * If `OfferChains` is present, this specifies which chain is going to be use to pay.
+   */
+  case class InvoiceRequestChain(hash: ByteVector32) extends InvoiceRequestTlv
+
+  /**
+   * Amount that the sender is going to send.
+   */
+  case class InvoiceRequestAmount(amount: MilliSatoshi) extends InvoiceRequestTlv
+
+  /**
+   * Features supported by the sender to pay the offer.
+   */
+  case class InvoiceRequestFeatures(features: Features[Feature]) extends InvoiceRequestTlv
+
+  /**
+   * Number of items to purchase. Only use if the offer supports purchasing multiple items at once.
+   */
+  case class InvoiceRequestQuantity(quantity: Long) extends InvoiceRequestTlv
+
+  /**
+   * A public key for which the sender know the corresponding private key.
```

Comment:
It would be useful to explain how this is used to complement the preimage proof of payment.